pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -853,7 +907,12 @@ cdef class _Timedelta(timedelta):
 
     def __hash__(_Timedelta self):
         if self._has_ns():
-            return hash(self.value)
+            if self._reso == NPY_FR_ns:
+                return hash(self.value)
+            else:
+                # TODO: more performant way to do this? same problem for Timestamp
+                # Invariances we are supposed to preserve?
```

Comment:
One is td1 == td2 -> hash(td1) == hash(td2) which I think this preserves. Would be good to add a test for it.

Reply:
the trouble is that two Timedeltas with different _reso can be equal, but with this implementation will have different hashes

Reply:
Ah I see. Since numpy doesn't hold this invariant I guess it's okay for Timedelta to not respect it too. Would be good to document it somewhere

```
In [7]: np.timedelta64(1, "D") == np.timedelta64(24, "h")
Out[7]: True

In [8]: hash(np.timedelta64(1, "D")) == hash(np.timedelta64(24, "h"))
Out[8]: False
```

Reply:
Will add comment with xref #44504.  can also simplify this to just return self.value at the cost of some hash collisions when we have two Timedeltas with the same .value but different ._reso, but I don't think that is likely to happen often in practice.